ambry-api/src/main/java/com/github/ambry/account/Account.java

```diff
@@ -306,12 +306,18 @@ public int hashCode() {
   }
 
   /**
-   * Adds a {@link Container} to this account and updates internal maps accordingly.
-   * @param container The container to update this account.
+   * Adds or removes a {@link Container} to this account and updates internal maps accordingly.
+   * @param container The container to be updated to this account.
+   * @param remove boolean which tells whether to add or remove the container.
    */
-  private void updateContainerMap(Container container) {
-    containerIdToContainerMap.put(container.getId(), container);
-    containerNameToContainerMap.put(container.getName(), container);
+  public void updateContainerMap(Container container, boolean remove) {
```

Comment:
I don't think we need this remove option.  When you are building the json for the account, you can do:
JSONObject jsonObj = account.toJson(false);
jsonObj.remove(CONTAINERS_KEY);
And use the compacted json in the DB record.
(You might need to add helper methods in AccountSerdeUtils.)